lib/irb/cmd/measure.rb

```diff
@@ -12,32 +12,29 @@ def initialize(*args)
         super(*args)
       end
 
-      def execute(type = nil, arg = nil, &block)
+      def execute(type = nil, arg = nil)
         # Please check IRB.init_config in lib/irb/init.rb that sets
         # IRB.conf[:MEASURE_PROC] to register default "measure" methods,
         # "measure :time" (abbreviated as "measure") and "measure :stackprof".
+
+        if block_given?
+          warn 'Configure IRB.conf[:MEASURE_PROC] to add custom measure methods.'
+          return
+        end
+
         case type
         when :off
-          IRB.conf[:MEASURE] = nil
           IRB.unset_measure_callback(arg)
         when :list
           IRB.conf[:MEASURE_CALLBACKS].each do |type_name, _, arg_val|
             puts "- #{type_name}" + (arg_val ? "(#{arg_val.inspect})" : '')
           end
         when :on
-          IRB.conf[:MEASURE] = true
-          added = IRB.set_measure_callback(type, arg)
+          added = IRB.set_measure_callback(nil, nil)
           puts "#{added[0]} is added." if added
         else
-          if block_given?
-            IRB.conf[:MEASURE] = true
```

Comment:
Changing `IRB.conf[:MEASURE]` is moved to `IRB.set_measure_calback` and `IRB.unset_measure_calback`.
This makes `IRB.set_measure_calback do end` to be a complete alternative of `measure do end`